packages/form-field/src/FormFieldInputContainer/FormFieldInputContainer.tsx

```diff
@@ -44,73 +37,50 @@ export const FormFieldInputContainer = forwardRef<
     const { disabled, size, state, inputProps, optional } =
       useFormFieldContext();
 
+    const inputWrapperStyles = getInputWrapperStyles({
+      disabled,
+      size: size ?? Size.Default,
+      state,
+      theme,
+    });
+
     const renderedChildren = React.cloneElement(children, {
       ...inputProps,
       className: cx(inputElementClassName, children.props.className),
     });
 
     const shouldRenderOptionalText =
       state === FormFieldState.None && !disabled && optional;
+    const shouldRenderIconWrapper = shouldRenderOptionalText || contentEnd;
```

Comment:
the name `shouldRenderIconWrapper` is a little confusing, does `optional` render an icon?